src/kaltura-live-plugin.tsx

```diff
@@ -109,16 +105,24 @@ export class KalturaLivePlugin
     onMediaUnload(): void {
         this.isMediaLive = false;
         this.isLiveMediaLoaded = false;
+    }
+
+    onPluginDestroy(): void {
         this._resetTimeout();
         this._player.removeEventListener(this._player.Event.FIRST_PLAY, this._handleFirstPlay);
-
         this._player.removeEventListener(this._player.Event.TIMED_METADATA, this.handleTimedMetadata);
-        this._player.removeEventListener(this._player.Event.ENDED, this._handleEnd);
-        this._player.removeEventListener(this._player.Event.ABORT, this._handleEnd);
+        this._player.removeEventListener(this._player.Event.ENDED, this._handleEndOrAbort);
+        this._player.removeEventListener(this._player.Event.ABORT, this._handleEndOrAbort);
     }
 
     private _activatePlugin = () => {
       this.isMediaLive = this.player.isLive();
+      if (this.isMediaLive) {
+        this._player.addEventListener(this._player.Event.FIRST_PLAY, this._handleFirstPlay);
+        this._player.addEventListener(this._player.Event.TIMED_METADATA, this.handleTimedMetadata);
+        this._player.addEventListener(this._player.Event.ENDED, this._handleEndOrAbort);
```

Comment:
ENDED and ABORT relevant after we set reload media on error? wouldn't it cause issues?
In our scenario, today, when the BE return an incorrect value, update reload media then check from BE and then get the ENDED after we reload the media will set the reload media again

Reply:
If we end stream - END event happens and not ERROR
so we need reload after Event.ENDED

Reply:
On chrome, we get ERROR first and then ENDED, that how it worked today when `reloadmedia` flag set only on ENDED/ABORT